gobject-introspection: fine-grained glib dependency versions. add 1.79.1

koysean opened the pull request:
gobject-introspection's glib dependency seems to require matching minor versions: https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/main/meson.build?ref_type=heads#L131-138

I don't know a good way to programatically do this so I explicitly added fine-grained glib dependencies.

I also updated both glib and gobject-introspection to their most recent release versions, 2.79.1 and 1.79.1, respectively.

koysean committed:
gobject-introspection: fine-grained glib dependency versions. add 1.79.1
glib: add 2.79.1

michaelkuhn commented on var/spack/repos/builtin/packages/glib/package.py:
```diff
@@ -28,6 +28,7 @@ class Glib(MesonPackage, AutotoolsPackage):
 
     license("LGPL-2.1-or-later")
 
+    version("2.79.1", sha256="b3764dd6e29b664085921dd4dd6ba2430fc19760ab6857ecfa3ebd4e8c1d114c")
```
This is not a release version but a development version. If there isn't an urgent reason to add this, we should skip it.

michaelkuhn commented on var/spack/repos/builtin/packages/gobject-introspection/package.py:
```diff
@@ -20,6 +20,7 @@ class GobjectIntrospection(MesonPackage, AutotoolsPackage):
 
     license("LGPL-2.0-or-later AND GPL-2.0-or-later AND MIT")
 
+    version("1.79.1", sha256="f80ea33ee05ca48fb997952bb46131cfbdd3751f7f5da068888739cbeb2d5443")
```
I guess the same is true for gobject-introspection's development versions.

michaelkuhn commented on var/spack/repos/builtin/packages/gobject-introspection/package.py:
```diff
@@ -44,6 +45,12 @@ class GobjectIntrospection(MesonPackage, AutotoolsPackage):
     depends_on("glib@2.49.2:", when="@1.49.2:")
     # version 1.48.0 build fails with glib 2.49.4
     depends_on("glib@2.48.1", when="@1.48.0")
+    depends_on("glib@2.49.2:", when="@1.49.2:")
+    depends_on("glib@2.56:", when="@1.56")
+    depends_on("glib@2.72:", when="@1.72")
+    depends_on("glib@2.76:", when="@1.76")
+    depends_on("glib@2.79:", when="@1.79")
+
```
These should be sorted like versions, that is, newest first. Also, do they have to be strict (`depends_on("glib@2.76", when="@1.76")`) or are newer glib versions ok?

koysean committed:
gobject-introspection: remove 1.79.1, add 1.78.1. reorder version numbers.
glib: remove 2.79.1

koysean commented:
@michaelkuhn thanks for catching the development version. I've removed 1.79.1 and added 1.78.1 instead.

The meson.build sets the requirement for glib version `>=` the gobject-introspection version:
`glib_version = '>=2.@0@.@1@'.format(gi_versions[1], glib_micro_version)`

michaelkuhn reviewed (approved):
LGTM, thanks!
